shared-module/usb_hid/__init__.c

```diff
@@ -260,24 +243,32 @@ void usb_hid_gc_collect(void) {
     gc_collect_ptr(hid_devices_tuple);
 
     // Mark possible heap pointers in the static device list as in use.
-    for (mp_int_t i = 0; i < num_hid_devices; i++) {
+    for (mp_int_t device_idx = 0; device_idx < num_hid_devices; device_idx++) {
+
         // Cast away the const for .report_descriptor. It could be in flash or on the heap.
         // Constant report descriptors must be const so that they are used directly from flash
         // and not copied into RAM.
-        gc_collect_ptr((void *)hid_devices[i].report_descriptor);
-        gc_collect_ptr(hid_devices[i].in_report_buffer);
-        gc_collect_ptr(hid_devices[i].out_report_buffer);
+        gc_collect_ptr((void *)hid_devices[device_idx].report_descriptor);
+
+        // Collect all the OUT report buffers for this device.
+        for (size_t id_idx = 0; id_idx < hid_devices[device_idx].num_report_ids; id_idx++) {
+            gc_collect_ptr(hid_devices[id_idx].out_report_buffers[id_idx]);
```

Comment:
Collect the in report buffers too.

Reply:
Whoops, yes, thanks. I had originally removed the IN buffers, but realized I still needed them.